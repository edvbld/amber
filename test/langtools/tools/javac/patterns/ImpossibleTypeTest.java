/*
 * @test /nodynamiccopyright/
 * @summary Ensure that in type test patterns, the predicate is not trivially provable false.
 * @compile/fail/ref=ImpossibleTypeTest.out -XDrawDiagnostics ImpossibleTypeTest.java
 */
public class ImpossibleTypeTest {

    public static void main(String[] args) {

        int in = 42;
        Integer i = 42;

        if (i __matches String s ) {
            System.out.println("Broken");
        }
        if (i __matches Undefined u ) {
            System.out.println("Broken");
        }
    }
}
