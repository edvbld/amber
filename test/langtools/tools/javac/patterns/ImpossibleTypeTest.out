ImpossibleTypeTest.java:13:13: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: java.lang.Integer, java.lang.String)
ImpossibleTypeTest.java:16:23: compiler.err.cant.resolve.location: kindname.class, Undefined, , , (compiler.misc.location: kindname.class, ImpossibleTypeTest, null)
2 errors
